executor/distsql.go

```diff
@@ -653,8 +663,11 @@ func (w *indexWorker) fetchHandles(ctx context.Context, result distsql.SelectRes
 }
 
 func (w *indexWorker) extractTaskHandles(ctx context.Context, chk *chunk.Chunk, idxResult distsql.SelectResult) (handles []int64, err error) {
-	handles = make([]int64, 0, w.batchSize)
-	for len(handles) < w.batchSize {
+	requiredNow := int(atomic.LoadInt64(&w.idxLookup.parentReqRows))
+	requiredNow = mathutil.Min(requiredNow, w.batchSize)
```

Comment:
maybe this `requiredNow` always equal to `requiredNow`?

```
because `w.idxLookup.parentReqRows <= maxChunkSize`
then `requiredNow <= w.idxLookup.parentReqRows`
and because `w.batchSize` inital with `maxChunkSize` and x2 for each call
so `w.batchSize >= maxChunkSize`

so requiredRow <= w.batchSize
```

Reply:
You are right, and now the strategy has been updated:
1. make all workers lazy start in the first call of  `Next` to get the first `RequiredRows` from parent;
2. init the `batchSize` to the first `RequiredRows`;
3. keep the `double extend strategy` and `IndexLookupSize `;

Then if the parent only needs a few records, the `batchSize` will be initialized to a small number instead of `MaxChunkSize` as before, then the executor can return more quickly than before.